adapters/repos/db/vector/hnsw/config_update.go

```diff
@@ -90,7 +90,15 @@ func (h *hnsw) UpdateUserConfig(updated schema.VectorIndexConfig) error {
 	atomic.StoreInt64(&h.efFactor, int64(parsed.DynamicEFFactor))
 	atomic.StoreInt64(&h.flatSearchCutoff, int64(parsed.FlatSearchCutoff))
 
-	h.cache.updateMaxSize(int64(parsed.VectorCacheMaxObjects))
+	if h.compressed {
+		h.compressedVectorsCache.updateMaxSize(int64(parsed.VectorCacheMaxObjects))
+	} else {
+		h.cache.updateMaxSize(int64(parsed.VectorCacheMaxObjects))
+	}
+	//ToDo: check atomic operation
+	if !h.compressed && parsed.Compressed {
```

Comment:
@abdelr are we able to run this in the background? With 250,000 vectors running locally I usually get a timeout when triggering from the api. The classification api runs in the background so we could use the same process.

Reply:
We need to come up with a proper process to run async tasks as part of #2359 anyway. I want the team to design this architecture in the current (v1.18) cycle, so we could build it in v1.19. In the meantime, we could have a couple of options:

- Make a really primitive async option that is limited to a single node. Basically just fire and forget, add a field to indicate when it's ready that can be polled. 
- Keep it synchronous for now and accept that you need a really big timeout (because HNSW-PQ is going to be considered experimental/beta)
- We could technically reuse the process from classifications, but that has issues, too (and I'd rather focus that effort on being a proper async task scheduler that we can reuse for all async things in the future).